src/p2p_sync.rs

```diff
@@ -85,21 +85,48 @@ where
         Ok(())
     }
 
-    /// Reads data sent from peer and decrypts it
-    pub fn read(&mut self) -> std::io::Result<Vec<u8>> {
+    /// Reads data from a stream decrypts it returning the data and len
+    pub fn read_len(&mut self) -> std::io::Result<(Vec<u8>, usize)> {
```

Comment:
no need for this function to be public, it could create confusion
```suggestion
    fn read_len(&mut self) -> std::io::Result<(Vec<u8>, usize)> {
```